### PR TITLE
Set up db schema and created models

### PR DESCRIPTION
**Pull request by Raphaeljunior**

### CODE REVIEW

File: Models/Courses.py [hunk 1]
```diff
@@ -0,0 +1,16 @@
+from sqlalchemy.ext.declarative import declarative_base
+from  sqlalchemy import Column, Integer, String
+
+
+Base = declarative_base()
+
+
+class course(Base):
```
Comment (alanhdu) [inline Models/Courses.py:8]: Classnames should be uppercase `Course` not `course`

File: Models/Courses.py [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+    __tablename__ = 'courses'
+    department_ids = Column(String)
+    name = Column(String)
+    id = Column(Integer)
+    number = Column(Integer)
+
+    def __repr__(self):
+        return str(self.name)
```
Comment (alanhdu) [inline Models/Courses.py:16]: This isn't a very good representation -- at the very least, the `__repr__` should include the type of the object.

File: Models/Courses.py [hunk 3]
```diff
@@ -0,0 +1,16 @@
+from sqlalchemy.ext.declarative import declarative_base
+from  sqlalchemy import Column, Integer, String
+
+
+Base = declarative_base()
+
+
+class course(Base):
+    __tablename__ = 'courses'
+    department_ids = Column(String)
```
Comment (alanhdu) [inline Models/Courses.py:10]: Why is this named `department_ids` (plural)? Should this be an array of strings?

File: Models/Departments.py [hunk 4]
```diff
@@ -0,0 +1,12 @@
+from sqlalchemy.ext.declarative import declarative_base
+from sqlalchemy import Column, Integer, String
+
+Base = declarative_base()
+
+class departmemt(Base):
```
Comment (alanhdu) [inline Models/Departments.py:6]: Uppercase

File: Models/Departments.py [hunk 5]
```diff
@@ -0,0 +1,12 @@
...
+    __tablename__ = 'departments'
+    name = Column(String)
+    id = Column(Integer, primary_key=True)
+
+    def __repr__(self):
+        return str(self.name)
```
Comment (alanhdu) [inline Models/Departments.py:12]: Include type of object

File: Models/Departments.py [hunk 6]
```diff
@@ -0,0 +1,12 @@
+from sqlalchemy.ext.declarative import declarative_base
+from sqlalchemy import Column, Integer, String
+
+Base = declarative_base()
+
+class departmemt(Base):
+    __tablename__ = 'departments'
+    name = Column(String)
+    id = Column(Integer, primary_key=True)
```
Comment (alanhdu) [inline Models/Departments.py:9]: Consider making the `name` the primary key.

File: Models/Professors.py [hunk 7]
```diff
@@ -0,0 +1,15 @@
+from sqlalchemy.ext.declarative import declarative_base
+from sqlalchemy import Column, String, Integer
+
+Base = declarative_base()
+
+class professor(Base):
```
Comment (alanhdu) [inline Models/Professors.py:6]: Upper case

File: templates/home.py [hunk 8]
```diff
@@ -0,0 +1,19 @@
+from sqlalchemy import create_engine
+from sqlalchemy.orm import sessionmaker
+from Models.Reviews import Review
+URI = "postgres://xeeqkeipvcflil:fead0f891152e8f657d7ee59ba256aa36c24fed629a0ee765b6a6aac6da2610f@ec2-54-225-236-102.compute-1.amazonaws.com:5432/d7mipabothqsm0"
+engine = create_engine(URI)
+engine.connect()
+Session = sessionmaker(bind=engine)
+session = Session()
```
Comment (alanhdu) [inline templates/home.py:8]: Again, consider using `flask_sqlalchemy` to handle database - request session syncing.

File: templates/home.py [hunk 9]
```diff
@@ -0,0 +1,19 @@
+from sqlalchemy import create_engine
+from sqlalchemy.orm import sessionmaker
+from Models.Reviews import Review
+URI = "postgres://xeeqkeipvcflil:fead0f891152e8f657d7ee59ba256aa36c24fed629a0ee765b6a6aac6da2610f@ec2-54-225-236-102.compute-1.amazonaws.com:5432/d7mipabothqsm0"
```
Comment (alanhdu) [inline templates/home.py:4]: Again:

> This is totally 🙅. Plaintext passwords to real machines on the cloud is a huge security problem. Please delete this line and change the password on the machine immediately.

> @Raphaeljunior

File: Models/Professors.py [hunk 10]
```diff
@@ -0,0 +1,15 @@
...
+    __tablename__ = 'professors'
+    nugget = Column(String)
```
Comment (alanhdu) [inline Models/Professors.py:8]: Why not make this an enum?

File: Models/Professors.py [hunk 11]
```diff
@@ -0,0 +1,15 @@
...
+    first_name = Column(String)
+    middle_name = Column(String)
+    last_name = Column(String)
```
Comment (alanhdu) [inline Models/Professors.py:11]: Why do we need to store these separately? Why not just store the professor's name as a single string?

See http://www.kalzumeus.com/2010/06/17/falsehoods-programmers-believe-about-names/

File: Models/Departments.py [hunk 12]
```diff
@@ -0,0 +1,12 @@
+from sqlalchemy.ext.declarative import declarative_base
+from sqlalchemy import Column, Integer, String
+
+Base = declarative_base()
```
Comment (alanhdu) [inline Models/Departments.py:4]: ...Does this work? I mean, you're creating a *different* base for each model. Why don't you create a common `models/__init__.py` which declares the base, and then `from . import Base` in each model file?

Also, why even bother breaking them up into multiple files? Seems premature to me.

**Comment by alanhdu** [comment]
@Raphaeljunior What's the code review process look like for y'all? The password thing in particularly should never have been merged in the first place.

File: Models/Courses.py (hunk 1 above)
Comment (Raphaeljunior) [reply Models/Courses.py:8]: Ok.

File: Models/Courses.py (hunk 3 above)
Comment (Raphaeljunior) [reply Models/Courses.py:10]: I don't think the database supports me to write arrays. Later on, I will wrap this and present it as a list.

File: Models/Courses.py (hunk 2 above)
Comment (Raphaeljunior) [reply Models/Courses.py:16]: I was testing something and forgot to comment it out. Ideally it should be something unique that can be used to hash the object instance.

File: Models/Departments.py (hunk 12 above)
Comment (Raphaeljunior) [reply Models/Departments.py:4]: Ok. Will do. Still learning flask. My bad.

File: Models/Departments.py (hunk 4 above)
Comment (Raphaeljunior) [reply Models/Departments.py:6]: Ok

File: Models/Departments.py (hunk 6 above)
Comment (Raphaeljunior) [reply Models/Departments.py:9]: Ok.

File: Models/Professors.py (hunk 11 above)
Comment (Raphaeljunior) [reply Models/Professors.py:11]: Thanks. Nice read.

File: templates/home.py (hunk 9 above)
Comment (Raphaeljunior) [reply templates/home.py:4]: Rookie move. How should I store the URI for database connections?

File: templates/home.py (hunk 8 above)
Comment (Raphaeljunior) [reply templates/home.py:8]: Will do. Still figuring out flask. Thanks

**Comment by Raphaeljunior** [comment]
@alanhdu the Merge was premature. I will update my code by the end of the weekend. Thanks for the reviews. This is very helpful since am still figuring out flask.

File: Models/Courses.py (hunk 3 above)
Comment (alanhdu) [reply Models/Courses.py:10]: Well, depends on the database. Postgres supports arrays natively and so does SQLAlchemy (http://docs.sqlalchemy.org/en/latest/core/type_basics.html#sqlalchemy.types.ARRAY)

File: Models/Courses.py (hunk 2 above)
Comment (alanhdu) [reply Models/Courses.py:16]: If you're looking for a unique hash, I'd just implement `__hash__` instead. IMO, `__repr__` is most used for debugging and logging (which is the difference between `__repr__` and `__str__` -- `__str__` is end-user facing, while `__repr__` is usually programmer / machine facing).

File: templates/home.py (hunk 9 above)
Comment (alanhdu) [reply templates/home.py:4]: Traditionally, we stored it in a config file that was `.gitignore`d and that we manually passed distributed around (although this is admittedly a bit of a PITA). The other way to do it is to have a `dev` environment where Postgres lives on the local machine (somewhere on `localhost`) and worry about a `prod` environment later.

There are lots of different ways to tackle this problem -- you should ask your mentor about what they think is best.

File: Models/Courses.py (hunk 1 above)
Comment (kmejia) [reply Models/Courses.py:8]: Interesting

**Pull request merged**